File: src/com/eichinn/composite/extension/Leaf.java
package com.eichinn.composite.extension;

/**
 * Created by ei_chinn on 2017/5/31.
 */
public class Leaf extends Component {
    private String name;

    public Leaf(String name) {
        this.name = name;
    }

    @Override
    public String getName() {
        return this.name;
    }

    @Override
    void printStruct(String preStr) {
        System.out.println(preStr + "-" + name);
    }
}

File: src/com/eichinn/visitor/extension/Leaf.kt
package com.eichinn.visitor.extension

/**
 * Created by ei_chinn on 2017/6/7.
 */
class Leaf(name: String) : Component(name) {
    override fun accept(visitor: Visitor) {
        visitor.visitLeaf(this)
    }

    override fun someOperation() {
        println("some operation in leaf")
    }

}
